nexus/src/external_api/http_entrypoints.rs

```diff
@@ -731,6 +737,134 @@ async fn saml_identity_provider_view(
 
 // TODO: no DELETE for identity providers?
 
+// "Local" Identity Provider
+
+/// List users in the given local identity provider
+#[endpoint {
+    method = GET,
+    path = "/system/silos/{silo_name}/identity-providers/local/users/all",
+    tags = ["system"],
+}]
+async fn local_idp_users_list(
+    rqctx: Arc<RequestContext<Arc<ServerContext>>>,
+    path_params: Path<SiloPathParam>,
+    query_params: Query<PaginatedById>,
+) -> Result<HttpResponseOk<ResultsPage<User>>, HttpError> {
+    let apictx = rqctx.context();
+    let nexus = &apictx.nexus;
+    let silo_name = path_params.into_inner().silo_name;
+    let query = query_params.into_inner();
+    let pagparams = data_page_params_for(&rqctx, &query)?;
+    let handler = async {
+        let opctx = OpContext::for_external_api(&rqctx).await?;
+        let users = nexus
+            .silo_api_users_list(&opctx, &silo_name, &pagparams)
+            .await?
+            .into_iter()
+            .map(|i| i.into())
+            .collect();
+        Ok(HttpResponseOk(ScanById::results_page(
+            &query,
+            users,
+            &|_, user: &User| user.id,
+        )?))
+    };
+    apictx.external_latencies.instrument_dropshot_handler(&rqctx, handler).await
+}
+
+/// Create a user
+///
+/// Users can only be created in Silos with `provision_type` == `Fixed`.
+/// Otherwise, Silo users are just-in-time (JIT) provisioned when a user first
+/// logs in using an external Identity Provider.
+#[endpoint {
+    method = POST,
+    path = "/system/silos/{silo_name}/identity-providers/local/users/create",
```

Comment:
See above.  The appeal of `POST .../users` is that it looks like a REST collection.  But it's already not that because of the other endpoints ("all" and "id" and the other stuff I mentioned that I expect to come later).  So I went with the pattern that I think we use for attaching disks (it's `POST .../disks/attach`).

Reply:
:+1: